src/map.rs

```diff
@@ -368,12 +434,27 @@ where
     }
 
     #[inline]
-    /// Obtains the value to which `key` is mapped and passes it through the closure `then`.
-    ///
+    /// Apply `then` to the mapping for `key` and get its result.
```

Comment:
I think we may want to remove this method altogether now, but that's for a different PR.